core/trino-main/src/main/java/io/trino/sql/planner/PlanFragmenter.java

```diff
@@ -418,6 +438,31 @@ public PlanNode visitTableFunctionProcessor(TableFunctionProcessorNode node, Rew
             return context.defaultRewrite(node, context.get());
         }
 
+        @Override
+        public PlanNode visitRemoteSource(RemoteSourceNode node, RewriteContext<FragmentProperties> context)
+        {
+            List<SubPlan> children = completedSubPlans.stream()
+                    .filter(subPlan -> node.getSourceFragmentIds().contains(subPlan.getFragment().getId()))
+                    .collect(toImmutableList());
```

Comment:
assert that entry was found in `completedSubPlans` for each id in `node.getSourceFragmentIds()`